ENIGMAsystem/SHELL/Universal_System/Resources/loading.cpp

```diff
@@ -124,14 +124,27 @@ namespace enigma
 
     //Initialize extensions
     enigma::extensions_initialize();
+    
 
     //Go to the first room
     if (enigma_user::room_count)
       enigma::game_start();
-    else {
-      enigma_user::window_default();
-      enigma_user::window_set_visible(true);
-    }
+
+    enigma_user::window_default(false);
+    // window sized by first room, can make visible now
+    enigma_user::window_set_visible(true);
+    // required for global game setting resizeable window
+    enigma_user::window_set_sizeable(isSizeable);
+    // required for global game setting borderless window
+    enigma_user::window_set_showborder(showBorder);
+    // required for global game setting fullscreen window
+    enigma_user::window_set_fullscreen(isFullScreen);
+    // don't ask me why this is necessary, but it 100% is
+    enigma_user::window_set_rectangle(windowX, windowY, windowWidth, windowHeight);
+    
+    // needed for DrawGUI event
+    enigma_user::screen_init();
+    enigma_user::screen_refresh();
```

Comment:
Delete these two lines, they are in the room switch already. Move your code to execute before we go to the first room.

Reply:
Another thing is, we will need the window visible much earlier in fact in the future. The reason? To render the game splash screen.

Reply:
![Screenshot_20190730_230009](https://user-images.githubusercontent.com/4379204/62180409-f3f73a00-b31d-11e9-8fed-04334a7d0b0b.png)

Reply:
notice the room size

Reply:
Now would you just look at that:
![Screenshot_20190730_230043](https://user-images.githubusercontent.com/4379204/62180427-05d8dd00-b31e-11e9-836e-086241819f99.png)

Reply:
Conclusion: ***STOP*** telling me to change shit for no fucking reason, only to discover what you told me to do breaks everything, and then jump around with a million different other ideas that don't work that you tell me to do for hours until by sheer luck you stumble across something that actually does what you wanted. I'm not your slave.

Reply:
Right, so what we need to do now is pull the size of the first room during `initialize_everything`. Just include the room header that allows you to do `room_get_width(0);` and etc to initialize the window dimensions.

Reply:
This could easily be solved 1 fucking way - don't review linux shit for however long you don't have a linux system. Let Josh be the one to do it, as he can actually test the stuff he wants done before asking me to add it, which you are in no position to do, and it is turning me into a person who simply does everything for you unpaid.

Reply:
PS: I'm going to continue to delete comments that start new tangents. We're not going to do the splash screen in this pull request, and I have heard your opinions on the matter. The last I recalled, you were interested in such an _optional_ feature, but if not I won't even waste my time.

Reply:
This is my precious time you are wasting. Either get a linux machine so you can review more competently and soberly, otherwise, leave it all to Josh.

Reply:
Ok, I won't review anymore of your pull requests. Josh can take over now.

Reply:
@JoshDreamland can you please mudge this puppy now

Reply:
> Right, so what we need to do now is pull the size of the first room during `initialize_everything`. Just include the room header that allows you to do `room_get_width(0);` and etc to initialize the window dimensions. Might want to use room_first though, not sure which is safer/safest.

The problem with this is that it doesn't take the use of views into account.

Reply:
Ah, that's correct, the views variables, which compute window size uses, haven't been loaded at this point.
https://github.com/enigma-dev/enigma-dev/blob/7fcb1ce05d926daabdd76c38ac808da583f8b287/ENIGMAsystem/SHELL/Universal_System/roomsystem.cpp#L224

I don't know then. You could split your code in two maybe, do all the initialize window settings in `initialize_everything` and then just handle initGame for the window size where you are now. That makes this problem a little trickier to solve.

Another way of going about it is to refactor compute window size to accept what room's view coordinates to use in the computation. Better yet, refactor compute window size to actually be a room function, so the window can ask for what size it should be given any room.

But now, since this is getting into a big refactor territory, I'm not really convinced in pushing you into doing it.